src/mango/src/mango_cursor_text.erl

```diff
@@ -203,19 +205,31 @@ handle_hit(CAcc0, Sort, Doc) ->
             CAcc2
     end.
 
-apply_user_fun(CAcc, Doc) ->
-    FinalDoc = mango_fields:extract(Doc, CAcc#cacc.fields),
+apply_user_fun(CAcc0, Doc) ->
     #cacc{
+        fields = Fields,
         user_fun = UserFun,
         user_acc = UserAcc,
-        execution_stats = Stats
-    } = CAcc,
-    Stats0 = mango_execution_stats:incr_results_returned(Stats),
-    case UserFun({row, FinalDoc}, UserAcc) of
-        {ok, NewUserAcc} ->
-            CAcc#cacc{user_acc = NewUserAcc, execution_stats = Stats0};
-        {stop, NewUserAcc} ->
-            throw({stop, CAcc#cacc{user_acc = NewUserAcc, execution_stats = Stats0}})
+        execution_stats = Stats,
+        documents_seen = Seen0
+    } = CAcc0,
+    FinalDoc = mango_fields:extract(Doc, Fields),
+    case sets:is_element(FinalDoc, Seen0) of
```

Comment:
we should only put the doc id into the set. storing the entire document is overkill. We also just need id, rather than id and rev, as we would not want to return the same doc id twice for different revs anyway.